Automattic/simplenote-ios
# Refactor SimplenoteActivityItemSource

### Fix
This is a quick PR to refactor the setup for SimplenoteActivityItemSource.  Previously the SimplenoteActivityItemSource  required a note to be passed in at initialization, this was only required so that the FileManager could access the simperiumID when writing the note content to a file.

In this PR I have changed it so that SimplenoteActivityItemSource accepts text and a filename for the destination file, decoupling SimplenoteActivityItemSource from FileManager and Note.  

Also, to WPiOS, FileManager will use a temporary directory instead of the uses documents directory

### Test
Should work the same as before, but just to make sure
1. run unit tests
2. open up a note and go to Info > Share and try a few different options other than Books and print
3. Make sure to share to Wordpress and save draft to Wordpress, there is custom logic in SimplenoteActivityItemSource for that

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer s required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.
